packages/miniflare/src/plugins/core/inspector-proxy/inspector-proxy-relay.ts

```diff
@@ -0,0 +1,161 @@
+import assert from "node:assert";
+import WebSocket from "ws";
+import { isDevToolsEvent } from "./devtools";
+import type {
+	DevToolsCommandRequests,
+	DevToolsEvent,
+	DevToolsEvents,
+} from "./devtools";
+
+export class InspectorProxyRelay {
+	#workerName: string;
+	#runtimeWs: WebSocket;
+
+	#devtoolsWs?: WebSocket;
+	#devtoolsHaveFileSystemAccess = false;
+
+	constructor(workerName: string, runtimeWs: WebSocket) {
+		this.#workerName = workerName;
+		this.#runtimeWs = runtimeWs;
+		this.#runtimeWs.once("open", () => this.#handleRuntimeWebSocketOpen());
+	}
+
+	get workerName() {
+		return this.#workerName;
+	}
+
+	get path() {
+		return `/${this.#workerName}`;
+	}
+
+	onDevtoolsConnected(
+		devtoolsWs: WebSocket,
+		devtoolsHaveFileSystemAccess: boolean
+	) {
+		/** We only want to have one active Devtools instance at a time. */
+		assert(
+			!this.#devtoolsWs,
+			"Too many clients; only one can be connected at a time"
+		);
+		this.#devtoolsWs = devtoolsWs;
+		this.#devtoolsHaveFileSystemAccess = devtoolsHaveFileSystemAccess;
+
+		assert(this.#devtoolsWs?.readyState === WebSocket.OPEN);
+
+		this.#devtoolsWs.on("error", console.error);
+
+		this.#devtoolsWs.once("close", () => {
+			if (this.#runtimeWs?.OPEN) {
+				// Since Miniflare proxies the inspector, reloading Chrome DevTools won't trigger debugger initialisation events (because it's connecting to an extant session).
+				// This sends a `Debugger.disable` message to the remote when a new WebSocket connection is initialised,
+				// with the assumption that the new connection will shortly send a `Debugger.enable` event and trigger re-initialisation.
+				// The key initialisation messages that are needed are the `Debugger.scriptParsed` events.
+				this.#sendMessageToRuntime({
+					method: "Debugger.disable",
+					id: this.nextCounter(),
+				});
+			}
+			this.#devtoolsWs = undefined;
+		});
+
+		this.#devtoolsWs.on("message", (data) => {
+			const msg = JSON.parse(data.toString());
+			assert(this.#runtimeWs?.OPEN);
+			this.#sendMessageToRuntime(msg);
+		});
+	}
+
+	#runtimeMessageCounter = 1e8;
```

Comment:
Out of interest, why not start at 1?

Reply:
I'm not actually sure, I simply followed the pattern already present in the inspector proxy worker:
https://github.com/cloudflare/workers-sdk/blob/5109fdd0bd5d57f22915fdf5c8114faf687cb0d2/packages/wrangler/templates/startDevWorker/InspectorProxyWorker.ts#L374

my guess is that this does distinguish our logs from standard/read ones coming from devtools that do start with 1... if you want I can change this to 1, I think it would work pretty much the same 🤔

Reply:
No don't change it. I was just curious.